mypy/checker.py

```diff
@@ -1254,6 +1254,24 @@ def visit_class_def(self, defn: ClassDef) -> None:
                 # Otherwise we've already found errors; more errors are not useful
                 self.check_multiple_inheritance(typ)
 
+            if defn.decorators:
```

Comment:
I would leave some kind of `TODO` here. One of main use cases:
```python
dec: Callable[[Type[A]], Type[B]]

@dec
class A:
    pass

def fun(arg: B) -> None:
    pass

fun(A())  # Should be OK
```
is still not supported.

Reply:
Added below